python/demo/demo-part3-final-exercise-solution.ipynb

```diff
@@ -0,0 +1,456 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Dataset in a Nutshell - Part 3: Neutron Data\n",
+    "\n",
+    " This is the continuation of [Dataset in a Nutshell - Part 2](demo-part2.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "import dataset as ds\n",
+    "from dataset import Dim, Coord, Data, Attr"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Converters from Mantid workspaces\n",
+    "\n",
+    "For convenience and testing, very basic converters from Mantid's `EventWorkspace` and `Workspace2D` to `Dataset` are available:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import mantid.simpleapi as mantid\n",
+    "import dataset.compat.mantid as mantidcompat\n",
+    "filename = 'PG3_4844_event.nxs'\n",
+    "filename_vanadium = 'PG3_4866_event.nxs'\n",
+    "\n",
+    "banks = []\n",
+    "banknames = ['bank124', 'bank144', 'bank164', 'bank184']\n",
+    "for bank in banknames:\n",
+    "    sampleWS = mantid.LoadEventNexus(filename, BankName=bank)\n",
+    "    vanadiumWS = mantid.LoadEventNexus(filename_vanadium, BankName=bank)\n",
+    "\n",
+    "    d = mantidcompat.to_dataset(sampleWS, name='sample', drop_pulse_times=False)\n",
+    "    d.merge(mantidcompat.to_dataset(vanadiumWS, name='vanadium', drop_pulse_times=True))\n",
+    "    banks.append(d)\n",
+    "    \n",
+    "d = banks[0]\n",
+    "for d2 in banks[1:]:\n",
+    "    d = ds.concatenate(d, d2, Dim.Row)\n",
+    "d[Coord.Row] = ([Dim.Row], [124,144,164,184])\n",
```

Comment:
I think I would add some comments here to make it clear why we are doing this. Adding this new coordinate was in the hints, so I would add a small comment explaining what we are doing here.
Also explain where the `arange(1078)` comes from.